packages/salesforcedx-vscode-core/test/jest/decorators/scratch-org-decorator.test.ts

```diff
@@ -0,0 +1,66 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import * as vscode from 'vscode';
+import { ORG_OPEN_COMMAND } from '../../../src';
+import { showOrg } from '../../../src/decorators';
+import { nls } from '../../../src/messages';
+
+describe('scratch org decorator', () => {
+  const testUser = 'test@username.com';
+  const browserIcon = `$(browser)`;
+  const openOrgCommand = ORG_OPEN_COMMAND;
+  let createFileSystemWatcherMock: jest.SpyInstance;
+  let getDefaultUsernameOrAliasMock: jest.SpyInstance;
+  let createStatusBarItemMock: jest.SpyInstance;
+  const mockStatusBarItem: any = {};
+  let mockWatcher: any;
+
+  beforeEach(() => {
+    mockStatusBarItem.tooltip = '';
+    mockStatusBarItem.command = '';
+    mockStatusBarItem.text = '';
+    mockStatusBarItem.show = jest.fn();
+    mockStatusBarItem.dispose = jest.fn();
+    mockWatcher = {
+      onDidChange: jest.fn(),
+      onDidCreate: jest.fn()
+    };
+    createStatusBarItemMock = (vscode.window.createStatusBarItem as any).mockReturnValue(
+      mockStatusBarItem
+    );
+    createFileSystemWatcherMock = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
+    getDefaultUsernameOrAliasMock = jest
+      .spyOn(ConfigUtil, 'getDefaultUsernameOrAlias');
+  });
+
+  describe('show Org', () => {
+    it('should show the browser icon in the status bar when a default username is set', async () => {
+      getDefaultUsernameOrAliasMock.mockResolvedValue(testUser);
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).toHaveBeenCalled();
+      expect(mockStatusBarItem.tooltip).toEqual(nls.localize('status_bar_open_org_tooltip'));
+      expect(mockStatusBarItem.command).toEqual(openOrgCommand);
+      expect(mockStatusBarItem.show).toHaveBeenCalled();
+      expect(mockStatusBarItem.text).toEqual(browserIcon);
+    });
+    it('should not show the browser icon in the status bar when a default username is not set', async () => {
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.tooltip).toEqual('');
+      expect(mockStatusBarItem.command).toEqual('');
+      expect(mockStatusBarItem.show).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.text).toEqual('');
+    });
+    it('should dispose and set to undefined the status bar when a default username is not set and the status bar exists', async () => {
+      mockStatusBarItem.text = 'text';
+      getDefaultUsernameOrAliasMock.mockResolvedValue(undefined);
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.dispose).toHaveBeenCalled();
+      expect(mockStatusBarItem).toEqual(undefined);
```

Comment:
I don't think this can be tested since we don't have a reference to the internal instance of statusBarItem after setting it to undefined.  

What if instead of setting it back to undefined, we use show and hide here instead:

<img width="500" alt="2022-11-11_14-54-59" src="https://user-images.githubusercontent.com/46458081/201441799-c766707d-3d4d-4b31-a1be-96535fd13d24.png">

This might be using show and hide as intended, and is easier to test since we can test that `hide()` was called.

Reply:
This test could then look like:
<img width="600" alt="2022-11-11_15-03-09" src="https://user-images.githubusercontent.com/46458081/201442285-aca97888-bef2-4838-9062-23ddf3f79b38.png">

Reply:
We considered using `.hide()` but thought `.dispose()` was better and is already calling `.hide()`. I think then maybe just testing that dispose was called is ok ?

Reply:
👀

Reply:
OK, I think this is fine as is.